RFCs/0000-expand-new-task.md

```diff
@@ -0,0 +1,158 @@
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ps_task.json' from template.
+```
+
+### Add `--from` option to `pdk new task`
+
+- The new `--from` option will require an argument which is an absolute or relative path to an existing script file.
+
+- If the `--from` option is specified without an argument, PDK will exit with an error.
+
+- If the given file does not exist or is not readable, PDK will exit with an error.
+
+- If the given file exists and is readable, PDK will copy it into the `tasks` subdirectory of the current module and write the copy with a name matching the task name supplied to `pdk new task`.
+
+  - For example: `pdk new task --from=/tmp/myscript.sh restart` will copy `/tmp/myscript.sh` to `tasks/restart.sh`.
+
+- For languages that support it (currently only PowerShell?), PDK will attempt to automatically determine the parameters (and output format?) of the source script and use that data to automatically populate the Bolt Task metadata file.
+
+### Add `--private` and `--remote` flags to `pdk new task`
+
+- When specified, these flags will simply pre-populate the generated Bolt Task metadata with `"private": true` and `"remote": true` key/value pairs respectively.
+
+### Add optional `<implementations>` argument to `pdk new task`
+
+- Allow `pdk new task` to accept an unbounded, space-separated list of `implementations` to be provided after the task name.
+
+- When one or more implementation files are provided, PDK will only generate a `tasks/<taskname>.json` file for the new task. The generated metadata will contain an `"implementations"` key with each given implementation file listed.
+
+- If the `--language` option is provided at the same time as one or more `<implementation>` arguments, PDK should exit with an appropriate error message.
+
+#### Examples
+
+The following example illustrates how a user might use PDK to author a new "cross-platform" task with Windows- and Linux-specific implementations:
+
+```bash
+$ pdk new task --private --language=bash sql_linux
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_linux.sh' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_linux.json' from template.
+
+$ pdk new task --private --language=powershell sql_windows
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_windows.ps1' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql_windows.json' from template.
+
+$ pdk new task sql tasks/sql_linux.sh tasks/sql_windows.ps1
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json' from template.
+pdk (INFO): Added implementation 'sql_linux.sh' to '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json'
+pdk (INFO): Added implementation 'sql_windows.rb' to '/Users/jesse/sandbox/pdk/testmod/tasks/sql.json'
+```
+
+Given the following task implementation script already exists at `tasks/init.rb`:
+
+```ruby
+#!/usr/bin/env ruby
```

Comment:
Should the default ruby shebang be `/usr/bin/env` or `/opt/puppetlabs/puppet/bin`?  I think most ruby tasks are really the latter but it might be a good idea to check forge.

Reply:
This was just an example (taken from the Bolt docs) of a shared implementation file that the user already has written, not necessarily a template-generated file.